sdk/identity/azure-identity/samples/azure/service/client.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/credentials/credentials.hpp>
+
+#include <memory>
+#include <utility>
+
+namespace Azure { namespace Service {
+
+  // This class is an oversimplified placeholder that is intended to represent an Azure SDK service
+  // client that uses Azure::Core::Credentials::TokenCredential to authenticate: Key Vault client,
+  // Storage Blobs client, etc.
+  class Client final {
+  private:
+    std::shared_ptr<Core::Credentials::TokenCredential> m_credential;
+
+  public:
+    explicit Client(
+        const std::string& someParameter,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential)
+        : m_credential(std::move(credential))
+    {
+      static_cast<void>(someParameter); // to suppress the "unused variable" warning.
```

Comment:
Would it be better for the sample to add some sort of usage?

Reply:
See below for more details, but here, it is better to have as less code as possible so that it does not draw any attention, the main demonstration is in other files.